### PR TITLE
Issue #427 - Dependencies to GitHub Packages are not included in PR handler

### PR DESCRIPTION
**Pull request by freddydk**
1. Update AL Go System files cannot downgrade due to missing Outputs under Initialization in old version.
2. PR handler needs access to GitHubPackagesContext in order to resolve dependencies

### CODE REVIEW

File: Actions/CheckForUpdates/CheckForUpdates.ps1 [hunk 1]
```diff
@@ -250,14 +250,16 @@ try {
                         if ($depth -gt 1) {
                             # When there are multiple build jobs, we need to add build job specific project list (and count) to the output of the Initialization job
                             $initializationOutputs = $yaml.Get('jobs:/Initialization:/outputs:/')
-                            $addOutput = @()
-                            1..$depth | ForEach-Object {
-                                $addOutput += @(
-                                  "projects$($_): `${{ steps.BuildOrder.outputs.projects$($_)Json }}"
-                                  "projects$($_)Count: `${{ steps.BuildOrder.outputs.projects$($_)Count }}"
-                                )
+                            if ($initializationOutputs) {
```
Comment (mazhelez) [inline Actions/CheckForUpdates/CheckForUpdates.ps1:253]: If there is no output, shouldn't the process create it?

**Review by mazhelez — APPROVED** [review]

File: Actions/CheckForUpdates/CheckForUpdates.ps1 (hunk 1 above)
Comment (freddydk) [reply Actions/CheckForUpdates/CheckForUpdates.ps1:253]: No, this is when downgrading - the old PR handler doesn't have the Outputs section.

**Comment by freddydk** [comment]
But... - there might be another problem
I have access to a partner repo, which uses GitHubPackages for dependency resolution, but in PR's we don't deliver to NuGet/GitHubPackages - and we shouldn't. We shouldn't deliver PR artifacts like that.
Will check this fix with their repo first.

File: Actions/CheckForUpdates/CheckForUpdates.ps1 (hunk 1 above)
Comment (freddydk) [reply Actions/CheckForUpdates/CheckForUpdates.ps1:253]: But the same problem exists with the build part, downgrading a repo now to test this

**Comment by freddydk** [comment]
> But... - there might be another problem I have access to a partner repo, which uses GitHubPackages for dependency resolution, but in PR's we don't deliver to NuGet/GitHubPackages - and we shouldn't. We shouldn't deliver PR artifacts like that. Will check this fix with their repo first.

No, there is no other known problem in this. For apps from the same repo we use the thisbuild artifacts - not github - this fix solved the issue from the partner.

**Review by aholstrup1 — APPROVED** [review]

**Pull request merged**